cli/src/main/java/com/devonfw/tools/ide/context/GitUrlSyntax.java

```diff
@@ -0,0 +1,95 @@
+package com.devonfw.tools.ide.context;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Enum representing the syntax of Git URLs, either SSH or HTTPS. Provides methods to format and convert Git URLs based on the syntax.
+ */
+public enum GitUrlSyntax {
+
+  /**
+   * The DEFAULT Git URL syntax
+   */
+  DEFAULT(""),
+  /**
+   * The SSH Git URL syntax (e.g., git@github.com:user/repo.git).
+   */
+  SSH("git@"),
+
+  /**
+   * The HTTPS Git URL syntax (e.g., https://github.com/user/repo.git).
+   */
+  HTTPS("https://");
+
+  private final String prefix;
+
+  private static final List<String> DOMAINS_WITH_NO_CONVERSION = Arrays.asList("github.com");
+
+  GitUrlSyntax(String prefix) {
+    this.prefix = prefix;
+  }
+
+  /**
+   * Formats the given Git URL according to the syntax represented by this enum constant.
+   * <p>
+   * Converts the URL between SSH and HTTPS formats. For example, an HTTPS URL can be converted to its corresponding SSH URL format, and vice versa.
+   * </p>
+   *
+   * @param gitUrl the original {@link GitUrl} to be formatted.
+   * @return the formatted {@link GitUrl} according to this syntax.
+   * @throws IllegalArgumentException if the protocol is not supported.
+   */
+  public GitUrl format(GitUrl gitUrl) {
+    String url = gitUrl.url();
+
+    // Prevent conversion for domains in the no-conversion list
+    if (isDomainWithNoConversion(url.toLowerCase())) {
+      return gitUrl;
+    }
+
+    switch (this) {
+      case DEFAULT:
```

Comment:
Why creating a new `GitUrl` instance if nothing changes?
For `DEFAULT` I would recommend to override this method this way:
```java
DEFAULT("") {
  @Override
  public GitUrl format(GitUrl gitUrl) {
    return gitUrl;
  }
}
```

You can also handle this here at the top:
```
public GitUrl format(GitUrl gitUrl) {
    if (this == DEFAULT) {
      return gitUrl;
    }
    String url = gitUrl.url();
    ...
}
```